### PR TITLE
Add unit test for search engine validation.

### PR DESCRIPTION
**Pull request by pocmo**

### CODE REVIEW

File: app/src/main/java/org/mozilla/focus/settings/ManualAddSearchEngineSettingsFragment.java [hunk 1]
```diff
@@ -196,7 +197,7 @@ private void showServerError(final ManualAddSearchEngineSettingsFragment that) {
     }
 
     @WorkerThread // makes network request.
-    private static boolean isValidSearchQueryURL(final String query) {
+    @VisibleForTesting static boolean isValidSearchQueryURL(final String query) {
```
Comment (sputnikci) [inline app/src/main/java/org/mozilla/focus/settings/ManualAddSearchEngineSettingsFragment.java]: \[PMD\] INFO: The method 'isValidSearchQueryURL' has a Cyclomatic Complexity of 5\.

Comment (pocmo) [reply app/src/main/java/org/mozilla/focus/settings/ManualAddSearchEngineSettingsFragment.java]: @sputnikci It wasn't me! 😝

File: app/src/test/java/org/mozilla/focus/settings/SearchEngineValidationTest.kt [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.mozilla.focus.settings.ManualAddSearchEngineSettingsFragment.isValidSearchQueryURL
+
+@RunWith(RobolectricTestRunner::class)
+// This unit test is not running on an Android device. Allow me to use spaces in function names.
+@Suppress("IllegalIdentifier")
+class SearchEngineValidationTest {
+    @Test
+    fun `URL returning 200 OK is valid`() = withMockWebServer(responseWithStatus(200)) {
```
Comment (mcomella) [inline app/src/test/java/org/mozilla/focus/settings/SearchEngineValidationTest.kt]: Interesting! It's pretty sweet we get one of the biggest benefits of behavioral unit tests (readability) without the libraries/boilerplate code.

File: app/src/main/java/org/mozilla/focus/settings/ManualAddSearchEngineSettingsFragment.java (hunk 1 above)
Comment (mcomella) [reply app/src/main/java/org/mozilla/focus/settings/ManualAddSearchEngineSettingsFragment.java]: Do you have a high level overview for what this warning means and how we should generally address it? I have the wikipedia page open for Cyclomatic Complexity and it looks non-trivial.

**Pull request merged**